src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,282 @@
+
+  const yInterpolate = floatingLabel.interpolate({
+    inputRange: [0, 1],
+    outputRange: [4, -15],
+  });
+
+  const animation = useMemo(() => {
+    return {
+      transform: [
+        {
+          translateY: yInterpolate,
+        },
+      ],
+    };
+  }, [yInterpolate]);
+
+  const moveLabelTop = () => {
+    Animated.timing(floatingLabel, {
+      toValue: 1,
+      duration: 200,
+      useNativeDriver: true,
+    }).start();
+  };
+
+  const moveLabelDown = () => {
+    Animated.timing(floatingLabel, {
+      toValue: 0,
+      duration: 200,
+      useNativeDriver: true,
+    }).start();
+  };
+
+  const onFocusHandler = (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+    moveLabelTop();
+    setFocused(true);
+    onFocus?.(e);
+  };
+
+  const onEndEditingHandler = (
+    e: NativeSyntheticEvent<TextInputEndEditingEventData>,
+  ) => {
+    if (e.nativeEvent.text?.length === 0 || value?.length === 0) {
+      moveLabelDown();
+    }
+    setFocused(false);
+    onEndEditing?.(e);
+  };
+
+  const TextInputStyles = useMemo(() => {
+    return [
+      styles.input,
+      {
+        borderBottomColor: colors.text,
+        color: colors.text,
+      },
+      fontStyles.b1_Text_Regular,
+      getStyle,
+      focused && mode !== 'border-less'
+        ? [styles.focused, {borderBottomColor: colors.border}]
+        : null,
+      error ? {borderColor: colors.error} : null,
+      getDisabledStyle,
+      inputStyle,
+    ];
+  }, [
+    colors.border,
+    colors.error,
+    colors.text,
+    error,
+    focused,
+    getDisabledStyle,
+    getStyle,
+    inputStyle,
+    mode,
+  ]);
+
+  const animatedLabelStyle = useMemo(() => {
+    return [
+      animation,
+      styles.animatedLabel,
+      mode === 'outline'
+        ? {
+            backgroundColor: colors.background,
+          }
+        : null,
+    ];
+  }, [animation, colors.background, mode]);
+
+  const showLabelStyle = useMemo(() => {
+    return [
+      {
+        backgroundColor:
+          !editable && mode === 'outline' ? colors.disabled : colors.background,
+        color: colors.text,
+      },
+      fontStyles.b3_Text_Italic,
+      focused ? [fontStyles.b3_Text_SemiBold, labelStyle] : null,
+    ];
+  }, [
+    colors.background,
+    colors.disabled,
+    colors.text,
+    editable,
+    focused,
+    labelStyle,
+    mode,
+  ]);
+
+  const animatedLabel = () => {
+    if (label) {
+      return (
+        <Animated.View style={animatedLabelStyle}>
+          <Label style={showLabelStyle} label={label} />
+        </Animated.View>
+      );
+    }
+    return null;
+  };
+
+  const inputField = () => {
+    return (
+      <InputField
+        style={TextInputStyles}
+        onFocus={onFocusHandler}
+        placeholder={placeholder}
+        editable={editable}
+        onChangeText={onChangeText}
+        onEndEditing={onEndEditingHandler}
+        {...restProps}
+      />
+    );
+  };
+
+  const hintAndError = () => {
```

Comment:
+1

Reply:
added render prefix for all the jsx elements.